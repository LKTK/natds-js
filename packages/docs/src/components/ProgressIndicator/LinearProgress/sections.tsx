import React from 'react';
import { LinearProgress } from '@naturacosmeticos/natds-web';

export const colors = [
  {
    title: 'Primary',
    component: <LinearProgress color="primary" />,
  },
  {
    title: 'Secondary',
    component: <LinearProgress color="secondary" />,
  },
];

export const variants = [
  {
    title: 'Determinate',
    component: (
      <LinearProgress color="primary" variant="determinate" value={70} />
    ),
  },
  {
    title: 'Indeterminate',
    component: <LinearProgress color="primary" variant="indeterminate" />,
  },
  {
    title: 'Buffer',
    component: (
      <LinearProgress
        color="primary"
        variant="buffer"
        valueBuffer={50}
        value={35}
      />
    ),
  },
  {
    title: 'Query',
    component: <LinearProgress color="primary" variant="query" value={75} />,
  },
];
