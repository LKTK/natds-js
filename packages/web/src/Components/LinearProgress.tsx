import React, { FunctionComponent, forwardRef } from 'react';
import { withTheme } from '@material-ui/styles';
import MaterialLinearProgress, {LinearProgressProps as ILinearProgressProps} from '@material-ui/core/LinearProgress';
export { LinearProgressProps as ILinearProgressProps } from '@material-ui/core/LinearProgress';

export const LinearProgress: FunctionComponent<Omit<ILinearProgressProps, 'translate'>> = forwardRef((
  props: Omit<ILinearProgressProps, 'translate'>,
  ref: any
) => (
  <MaterialLinearProgress {...props} ref={ref} />
));

export default withTheme(LinearProgress);
