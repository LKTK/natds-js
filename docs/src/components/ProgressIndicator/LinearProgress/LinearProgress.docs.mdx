import { LinearProgress, Provider, themes } from '@naturacosmeticos/natds-web';
import ComponentWithLabel from '../../../../addons/ComponentWithLabel';
import { colors, variants } from './sections';

# LinearProgress

---

---

## Colors

<Provider theme={themes.natura.light}>
  <ComponentWithLabel componentList={colors} itemsPerRow={1} />
</Provider>

## Variants

<Provider theme={themes.natura.light}>
  <ComponentWithLabel componentList={variants} itemsPerRow={1} />
</Provider>

---

---

## API

```jsx highlight-line="2"
import { LinearProgress } from '@naturacosmeticos/natds-web';

<LinearProgress color="primary" variant="indeterminate" value="70" />;
```

For more advanced usages, check [Material UI Progress docs](https://material-ui.com/pt/components/progress/)

---

---

## Circular Progress Props

| Name        |                          Type                           |     Default     |                                                                            Description                                                                             |
| :---------- | :-----------------------------------------------------: | :-------------: | :----------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| classes     |                         object                          |                 | Override or extend the styles applied to the component. See [Material Linear Progress CSS API](https://material-ui.com/api/linear-progress/#css) for more details. |
| color       |          'primary' \| 'secondary' \| 'inherit'          |    'primary'    |                                   The color of the component. It supports those theme colors that make sense for this component.                                   |
| value       |                         number                          |                 |                               The value of the progress indicator for the determinate and static variants. Value between 0 and 100.                                |
| valueBuffer |                         number                          |                 |                                                     The value for the buffer variant. Value between 0 and 100.                                                     |
| variant     | 'determinate' \| 'indeterminate' \| 'buffer' \| 'query' | 'indeterminate' |                                               The variant to use. Use indeterminate when there is no progress value.                                               |
