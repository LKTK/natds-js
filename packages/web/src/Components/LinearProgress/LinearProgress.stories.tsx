import { Meta, Story } from "@storybook/react";
import { ILinearProgressProps, LinearProgress } from "./LinearProgress";
import { colors } from "./__fixtures__/colors";
import { variants } from "./__fixtures__/variants";
import { Template, DynamicTemplate } from "./Template";

export default {
  component: LinearProgress,
  title: "Components/Linear Progress Indicator",
} as Meta;

export const Playground : Story<ILinearProgressProps> = Template.bind({});
Playground.args = {
  color: colors.primary,
  value: 50,
  valueBuffer: 70,
  variant: variants.indeterminate,
};

export const PrimaryColor : Story<ILinearProgressProps> = Template.bind({});
PrimaryColor.args = {
  color: colors.primary,
};

export const SecondaryColor : Story<ILinearProgressProps> = Template.bind({});
SecondaryColor.args = {
  color: colors.secondary,
};

export const DeterminateVariant : Story<ILinearProgressProps> = Template.bind({});
DeterminateVariant.args = {
  color: colors.primary,
  value: 70,
  variant: variants.determinate,
};

export const IndeterminateVariant : Story<ILinearProgressProps> = Template.bind({});
IndeterminateVariant.args = {
  color: colors.primary,
  value: null,
  variant: variants.indeterminate,
};

export const BufferVariant : Story<ILinearProgressProps> = Template.bind({});
BufferVariant.args = {
  color: colors.primary,
  value: 35,
  valueBuffer: 50,
  variant: variants.buffer,
};

export const QueryVariant : Story<ILinearProgressProps> = Template.bind({});
QueryVariant.args = {
  color: colors.primary,
  value: null,
  variant: variants.query,
};

export const DeterminateLiveExample : Story<ILinearProgressProps> = DynamicTemplate.bind({});
DeterminateLiveExample.args = {
  color: colors.primary,
  value: 0,
  variant: variants.determinate,
};
DeterminateLiveExample.parameters = {
  chromatic: { pauseAnimationAtEnd: true },
};

export const BufferLiveExample : Story<ILinearProgressProps> = DynamicTemplate.bind({});
BufferLiveExample.args = {
  color: colors.primary,
  value: 0,
  valueBuffer: 10,
  variant: variants.buffer,
};
BufferLiveExample.parameters = {
  chromatic: { pauseAnimationAtEnd: true },
};
