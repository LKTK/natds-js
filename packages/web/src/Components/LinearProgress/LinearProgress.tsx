import * as React from "react";
import MaterialLinearProgress from "@material-ui/core/LinearProgress";
import { ILinearProgressProps } from "./LinearProgress.props";

export { ILinearProgressProps } from "./LinearProgress.props";

/**
 * ## Importing
 *
 * ```
 * import { LinearProgress } from "@naturacosmeticos/natds-web";
 * ```
 */
export const LinearProgress = React.forwardRef<unknown, ILinearProgressProps>(
  (props: Omit<ILinearProgressProps, "translate">, ref) => (
    <MaterialLinearProgress {...props} ref={ref} />),
);

LinearProgress.displayName = "LinearProgress";

export default LinearProgress;
