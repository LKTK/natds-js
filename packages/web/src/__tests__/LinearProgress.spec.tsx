import React from 'react';
import { mount } from 'enzyme';
import { expect } from 'chai';
import renderer from 'react-test-renderer';

import LinearProgress from '../Components/LinearProgress';

describe('Circular Progress components', () => {
  describe('rendering variants', () => {
    test('should match to snapshot - empty LinearProgress', () => {
      const component = mount(<LinearProgress />);

      expect(component).matchSnapshot('LinearProgress empty snapshot');
    });

    test('should match snapshot - with variant indeterminate', () => {
      const component = renderer.
        create(<LinearProgress variant="indeterminate" />).
        toJSON();

      expect(component).to.matchSnapshot('With variant indeterminate snapshot');
    });

    test('should match snapshot - with variant determinate', () => {
      const component = renderer.
        create(<LinearProgress variant="determinate" value={70} />).
        toJSON();

      expect(component).to.matchSnapshot('With variant determinate snapshot');
    });

    test('should match snapshot - with variant Buffer', () => {
      const component = renderer.
        create(<LinearProgress variant="buffer" value={50} valueBuffer={70} />).
        toJSON();

      expect(component).to.matchSnapshot('With variant buffer snapshot');
    });

    test('should match snapshot - with variant query', () => {
      const component = renderer.
        create(<LinearProgress variant="query" />).
        toJSON();

      expect(component).to.matchSnapshot('With variant query snapshot');
    });
  });
});
