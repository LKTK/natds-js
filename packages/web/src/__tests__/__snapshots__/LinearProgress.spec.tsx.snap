// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circular Progress components rendering variants should match snapshot - with variant Buffer: With variant buffer snapshot 1`] = `
<div
  aria-valuenow={50}
  className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-buffer"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-dashed MuiLinearProgress-dashedColorPrimary"
  />
  <div
    className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Buffer"
    style={
      Object {
        "transform": "translateX(-50%)",
      }
    }
  />
  <div
    className="MuiLinearProgress-bar MuiLinearProgress-colorPrimary MuiLinearProgress-bar2Buffer"
    style={
      Object {
        "transform": "translateX(-30%)",
      }
    }
  />
</div>
`;

exports[`Circular Progress components rendering variants should match snapshot - with variant determinate: With variant determinate snapshot 1`] = `
<div
  aria-valuenow={70}
  className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
    style={
      Object {
        "transform": "translateX(-30%)",
      }
    }
  />
</div>
`;

exports[`Circular Progress components rendering variants should match snapshot - with variant indeterminate: With variant indeterminate snapshot 1`] = `
<div
  className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
    style={Object {}}
  />
  <div
    className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
    style={Object {}}
  />
</div>
`;

exports[`Circular Progress components rendering variants should match snapshot - with variant query: With variant query snapshot 1`] = `
<div
  className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-query"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
    style={Object {}}
  />
  <div
    className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
    style={Object {}}
  />
</div>
`;

exports[`Circular Progress components rendering variants should match to snapshot - empty LinearProgress: LinearProgress empty snapshot 1`] = `
<WithTheme(ForwardRef)>
  <ForwardRef>
    <WithStyles(ForwardRef(LinearProgress))>
      <ForwardRef(LinearProgress)
        classes={
          Object {
            "bar": "MuiLinearProgress-bar",
            "bar1Buffer": "MuiLinearProgress-bar1Buffer",
            "bar1Determinate": "MuiLinearProgress-bar1Determinate",
            "bar1Indeterminate": "MuiLinearProgress-bar1Indeterminate",
            "bar2Buffer": "MuiLinearProgress-bar2Buffer",
            "bar2Indeterminate": "MuiLinearProgress-bar2Indeterminate",
            "barColorPrimary": "MuiLinearProgress-barColorPrimary",
            "barColorSecondary": "MuiLinearProgress-barColorSecondary",
            "buffer": "MuiLinearProgress-buffer",
            "colorPrimary": "MuiLinearProgress-colorPrimary",
            "colorSecondary": "MuiLinearProgress-colorSecondary",
            "dashed": "MuiLinearProgress-dashed",
            "dashedColorPrimary": "MuiLinearProgress-dashedColorPrimary",
            "dashedColorSecondary": "MuiLinearProgress-dashedColorSecondary",
            "determinate": "MuiLinearProgress-determinate",
            "indeterminate": "MuiLinearProgress-indeterminate",
            "query": "MuiLinearProgress-query",
            "root": "MuiLinearProgress-root",
          }
        }
      >
        <div
          className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
          role="progressbar"
        >
          <div
            className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
            style={Object {}}
          />
          <div
            className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
            style={Object {}}
          />
        </div>
      </ForwardRef(LinearProgress)>
    </WithStyles(ForwardRef(LinearProgress))>
  </ForwardRef>
</WithTheme(ForwardRef)>
`;
